service/src/test/java/uk/gov/hmcts/reform/fpl/service/GeneratedOrderServiceTest.java

```diff
@@ -300,18 +305,28 @@ void shouldRemoveOrderPropertiesWhenTheyExistInCaseDetails() {
                 }
                 break;
             case SUPERVISION_ORDER:
-                final String suffix = dateFormatterService.getDayOfMonthSuffix(date.getDayOfMonth());
-                final String formattedDateTime =
-                    dateFormatterService.formatLocalDateTimeBaseUsingFormat(date.plusMonths(5),
-                        "h:mma 'on the' d'" + suffix + "' MMMM y");
                 expectedMap
-                    .put("orderType", SUPERVISION_ORDER)
-                    .put("orderTitle", "Supervision order")
-                    .put("childrenAct", "Section 31 and Paragraphs 1 and 2 Schedule 3 Children Act 1989")
-                    .put("orderDetails",
-                        String.format(
-                            "It is ordered that Example Local Authority supervises the child for 5 months from the "
-                                + "date of this order until %s.", formattedDateTime));
+                    .put("orderType", SUPERVISION_ORDER);
+                if (subtype == INTERIM) {
+                    expectedMap
+                        .put("orderTitle", "Interim supervision order")
+                        .put("childrenAct", "Section 38 and Paragraphs 1 and 2 Schedule 3 Children Act 1989")
+                        .put("orderDetails",
+                            "It is ordered that Example Local Authority supervises the child until the end of the "
+                                + "proceedings");
+                } else if (subtype == FINAL) {
+                    final String suffix = dateFormatterService.getDayOfMonthSuffix(date.getDayOfMonth());
+                    final String formattedDateTime =
+                        dateFormatterService.formatLocalDateTimeBaseUsingFormat(date.plusMonths(5),
+                            "h:mma 'on the' d'" + suffix + "' MMMM y");
+                    expectedMap
+                        .put("orderTitle", "Supervision order")
+                        .put("childrenAct", "Section 31 and Paragraphs 1 and 2 Schedule 3 Children Act 1989")
+                        .put("orderDetails",
+                            String.format(
+                                "It is ordered that Example Local Authority supervises the child for 5 months from the "
+                                    + "date of this order until %s.", formattedDateTime));
```

Comment:
This is no longer as clean as it once was, maybe extracting the logic within the if conditions may help?

Reply:
as per lewis comment I will remove the `else if (type == FINAL)`, but don't think it's worth extracting the logic

Reply:
Not removing the `else if` anymore as better to be safer, just on the off chance the enum is extended and we add more types. Still dont think its worth extracting the condition logic